app/uk/gov/hmrc/gform/testonly/SnapshotADTs.scala

```diff
@@ -16,78 +16,166 @@
 
 package uk.gov.hmrc.gform.testonly
 
-import uk.gov.hmrc.mongo.cache.CacheItem
 import java.time.Instant
 import julienrf.json.derived
 import play.api.libs.json.Reads._
 import play.api.libs.json._
+import uk.gov.hmrc.crypto.{ Decrypter, Encrypter }
+import uk.gov.hmrc.gform.save4later.EncryptedFormFormat
+import uk.gov.hmrc.gform.sharedmodel.form.Form
+import uk.gov.hmrc.gform.sharedmodel.form.FormData
+import uk.gov.hmrc.gform.sharedmodel.form.FormId
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.FormTemplateId
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.FormTemplateRaw
 
 case class Snapshot(
-  templateId: String,
-  snapshotId: String,
+  templateId: FormTemplateId,
+  snapshotId: SnapshotId,
   savedAt: Instant,
-  description: String
+  description: Description,
+  gformVersion: GformVersion,
+  gformFrontendVersion: GformFrontendVersion
 )
 
 object Snapshot {
 
-  def apply(cacheItem: CacheItem): Snapshot = {
-    val templateId = cacheItem.data.validate((__ \ "form" \ "formTemplateId").json.pick[JsString]).get.value
-    val snapshotId = cacheItem.id
-    val savedAt = cacheItem.createdAt
-    val description = cacheItem.data.validate((__ \ "description").json.pick[JsString]).get.value
+  def apply(snapshotItem: SnapshotItem): Snapshot =
     Snapshot(
-      templateId,
-      snapshotId,
-      savedAt,
-      description
+      snapshotItem.snapshotTemplateId,
+      snapshotItem.snapshotId,
+      snapshotItem.createdAt,
+      snapshotItem.description,
+      snapshotItem.gformVersion,
+      snapshotItem.gformFrontendVersion
     )
-  }
 
   implicit val writes: OWrites[Snapshot] = derived.owrites()
-
 }
 
 case class SnapshotWithData(
   snapshot: Snapshot,
-  formData: JsObject
+  formData: FormData
 )
 object SnapshotWithData {
```

Comment:
Why we need this type? Why not use `SnapshotItem` instead?

Reply:
yes, `SnapshotItem` should work here I think. i will refactor

Reply:
but again the SnapshotItem has template and the whole form, but for this I need only some meta information about snapshot, which is `Snapshot` and formdata

Reply:
Let's call this `SnapshotOverview`